include/swift/IRGen/Linking.h

```diff
@@ -10,22 +10,31 @@
 //
 //===----------------------------------------------------------------------===//
 
-#ifndef SWIFT_IRGEN_LINKENTITY_H
-#define SWIFT_IRGEN_LINKENTITY_H
+#ifndef SWIFT_IRGEN_LINKING_H
+#define SWIFT_IRGEN_LINKING_H
 
-#include "swift/AST/Types.h"
 #include "swift/AST/Decl.h"
 #include "swift/AST/ProtocolConformance.h"
+#include "swift/AST/Types.h"
 #include "swift/IRGen/ValueWitness.h"
-#include "swift/SIL/SILModule.h"
 #include "swift/SIL/SILFunction.h"
 #include "swift/SIL/SILGlobalVariable.h"
+#include "swift/SIL/SILModule.h"
 #include "llvm/ADT/DenseMapInfo.h"
+#include "llvm/IR/GlobalValue.h"
 
 namespace swift {
 namespace irgen {
 class IRGenModule;
 
+class UniversalLinkageInfo {
```

Comment:
I could flip the helper constructor of this around to `IRGenModule::getUniversalLinkageInfo()` or something, I suppose, and remove the `LinkInfo::get` helper overload, but I'm not entirely sure that's worth it. (Also, `isAvailableExternally` seemed like it would become far more ugly to become independent of `IRGenModule`: the argument would be replaced by a `DeclContext*` meant represent the module-scope of the `IRGenModule`.)